grails-app/services/au/org/ala/ecodata/ProjectService.groovy

```diff
@@ -735,50 +746,48 @@ class ProjectService {
+        int ignoredProjects = 0, createdProjects = 0, updatedProjects = 0, page = 1
+        JsonSlurper slurper = new JsonSlurper()
+
         log.info("Starting SciStarter import")
         try {
-            JsonSlurper jsonSlurper = new JsonSlurper()
-            String sciStarterProjectUrl
-            // list all SciStarter projects
-            List projects = getSciStarterProjectsFromFinder()
-            projects?.eachWithIndex { pProperties, index ->
-                Map transformedProject
-                Map project = pProperties
-                if (project && project.title && project.id) {
-                    Project importedSciStarterProject = Project.findByExternalIdAndIsSciStarter(project.id?.toString(), true)
-                    // get more details about the project
-                    try {
-                        sciStarterProjectUrl = "${grailsApplication.config.getProperty('scistarter.baseUrl')}${grailsApplication.config.getProperty('scistarter.projectUrl')}/${project.id}?key=${grailsApplication.config.getProperty('scistarter.apiKey')}"
-                        String text = webService.get(sciStarterProjectUrl, false);
-                        if (text instanceof String) {
-                            Map projectDetails = jsonSlurper.parseText(text)
-                            if (projectDetails.origin && projectDetails.origin == 'atlasoflivingaustralia') {
-                                // ignore projects SciStarter imported from Biocollect
-                                log.warn("Ignoring ${projectDetails.title} - ${projectDetails.id} - This is an ALA project.")
-                                ignoredProjects++
-                            } else {
-                                projectDetails << project
-                                // map properties from SciStarter to Biocollect
-                                transformedProject = SciStarterConverter.convert(projectDetails)
-                                if (!importedSciStarterProject) {
-                                    // create project & document & site & organisation
-                                    createSciStarterProject(transformedProject, projectDetails)
-                                    createdProjects++
-                                } else {
-                                    // update a project just in case something has changed.
-                                    updateSciStarterProject(transformedProject, importedSciStarterProject)
-                                    log.info("Updating ${importedSciStarterProject.name} ${importedSciStarterProject.projectId}.")
-                                    updatedProjects++
-                                }
-                            }
-                        }
-                    } catch (Exception e) {
-                        log.error("Error processing project - ${sciStarterProjectUrl}. Ignoring it. ${e.message}", e);
+
+            while(true) {
```

Comment:
The removal of the try/catch here I assume is deliberate?  Assuming sci-starter data is always well formatted etc which is probably OK?  Unfortunately generally our mongo setup will commit/flush each update so if one fails you'll still have all the earlier ones in the database but then stop processing.

Reply:
The SciStarter API output is pretty formatted & consistent - however I've re-implemented the try-catch to be safe